lib/scope.js

```diff
@@ -1086,15 +928,57 @@ module.exports = {
           }
 
           if ( rows_match ) {
-            matches.push({ field: field, table: table_row });
+            // Collect all the matches for each individual field - we
+            // may need to de-duplicate them.
+            if ( matches_by_field[ field.selector ] ) {  // selector should be unique per field
+              matches_by_field[ field.selector ].push({ field: field, table: table_row });
+            } else {
+              matches_by_field[ field.selector ] = [{ field: field, table: table_row }];
+            }
+
             continue;
           }
 
         }  // ends for each table row
       }  // ends for each field row possibility
     }  // ends for each field
 
-    return matches;
+    // Some fields will have multiple matches from the story table.
+    // We think this should only happen if the row is using a proxy vars.
+    // When a dev has a loop with an index var or generic object,
+    // the same variable name will be used over and over, e.g. `users[i]`.
+    // That's what will be in the story table, so there will be multiple
+    // matches for a field setting `users[i]`. For this program to
+    // know which row to use when, we need the one unique var that
+    // will appear on the page - the sought var. When a row with `users[i]`
+    // has a sought var described that matches the page sought var, the
+    // program knows to use that row to set the value.
+
+    // For now, we're being flexible about it - if we don't think we
+    // need that sought var because the situation is not as described
+    // above, we don't prevent data getting through.
+    // e.g. If an interview has a list of `users`, but they only set
+    // data for `users[0]`, there's no need to check the sought var.
+    // That way devs with simple cases can ignore the sought var column.
+    de_duped_matches = [];
+    for ( let field_key in matches_by_field  ) {
+      let field_matches = matches_by_field[ field_key ];
+      if ( page_data.uses_proxies && field_matches.length > 1 ) {
+        // If multiple story table rows match one field, only
+        // keep the one(s) that match this particular page (sought var)
+        for ( let match of field_matches ) {
+          if ( match.table.sought === page_data.sought_var ) {
+            de_duped_matches.push( match );
+          }
+        }
+      } else {
+        // If proxy vars aren't relevant here or if only
+        // one table row matched this field, use that match obj
+        de_duped_matches.push( field_matches[0] );
```

Comment:
Should we add all matches here? Not just the first one? I don't know when that would happen in legit code, but maybe it could and we're more likely to see that at some point if we don't hide it here. I can definitely see a dev accidentally adding double rows. We should therefore `concat()` and report if there are multiple matches. This will also make things more transparent to us, so it's an awesome idea.

Reply:
Made #262.